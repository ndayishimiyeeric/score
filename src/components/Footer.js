import React from 'react';
import styles from './styles/Footer.module.css';

function Footer() {
  return (
    <footer>
      <p className={styles.footerP}>
        &copy;
        decathlon.com
      </p>
    </footer>
  );
}

export default Footer;
