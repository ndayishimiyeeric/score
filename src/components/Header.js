import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { MdKeyboardArrowLeft, MdKeyboardVoice } from 'react-icons/md';
import { TbSettings } from 'react-icons/tb';
import styles from './styles/Header.module.css';

const Header = (props) => {
  const { heading } = props;
  return (
    <header className={styles.header}>
      <Link to="/" className={styles.link}>
        <MdKeyboardArrowLeft
          className={styles.arrowBack}
        />
      </Link>
      <p className={styles.headerHeading}>{heading}</p>
      <div className={styles.headerSide}>
        <MdKeyboardVoice
          className={styles.sideIcon}
        />
        <TbSettings
          className={styles.sideIcon}
        />
      </div>
    </header>
  );
};

export default Header;

Header.propTypes = {
  heading: PropTypes.string,
};

Header.defaultProps = {
  heading: 'header Heading',
};
