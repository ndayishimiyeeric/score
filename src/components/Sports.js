/* eslint-disable object-curly-newline */
/* eslint-disable function-paren-newline */
/* eslint-disable comma-dangle */
/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable newline-per-chained-call */
import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getData } from '../redux/sportsSlice/sportsSlice';
import Sport from './Sport';
import Header from './Header';
import Footer from './Footer';
import styles from './styles/Leagues.module.css';

const Sports = () => {
  const [formData, setFormData] = useState({
    textValue: '',
  });

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getData());
  }, []);

  const handleChange = (event) => {
    const { value, name, type, checked } = event.target;
    setFormData((prevFormData) => ({
      ...prevFormData,
      [name]: type === 'checkbox' ? checked : value,
    }));
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (formData.textValue.trim()) {
      setFormData((prevState) => ({
        ...prevState,
        textValue: '',
      }));
    }
  };

  const sports = useSelector((state) => state.sports);
  const changer = (str) =>
    str.toString().split('').join('').toUpperCase().replace(/\s/g, '');
  const NewsportElements = sports.filter((item) =>
    changer(item.name).includes(changer(formData.textValue))
  );

  const sportElements = NewsportElements.map((sport) => (
    <Sport
      key={sport.id}
      id={sport.id}
      img={sport.icon}
      title={sport.name}
      games={sport.tags}
    />
  ));
  const sportsAvailable = sports.filter((sport) => sport.icon !== null);

  return (
    <>
      <Header heading="Sports" />
      <div className={styles.mainDiv}>
        <div className={styles.showcaseDiv}>
          <div className={styles.showcaseImg}>
            <img
              src="https://sports-api-production.s3.amazonaws.com/uploads/sport/icon/224/224.svg"
              alt="showcase"
            />
          </div>
          <div className={styles.showcaseDetails}>
            <p>SPORTS</p>
            <span>
              {sportsAvailable.length}
              &nbsp; games
            </span>
          </div>
        </div>
        <p className={styles.heading}>SPORTS BY GROUP</p>
        <form onSubmit={handleSubmit}>
          <input
            type="text"
            placeholder="Search..."
            onChange={handleChange}
            name="textValue"
            value={formData.textValue}
            className={styles.formInput}
          />
        </form>
        {NewsportElements.length > 0 ? (
          <div className={styles.leaguesDiv}>{sportElements}</div>
        ) : (
          <div className={styles.missing}>
            <p>No such game available</p>
          </div>
        )}
      </div>
      <Footer />
    </>
  );
};

export default Sports;
