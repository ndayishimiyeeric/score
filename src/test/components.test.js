import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Sports from '../components/Sports';
import Footer from '../components/Footer';
import Header from '../components/Header';

describe('Test Compontests', () => {
  test('test Sports', () => {
    const tree = renderer
      .create(<Provider store={store}><BrowserRouter><Sports /></BrowserRouter></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('test Header', () => {
    const tree = renderer
      .create(<Provider store={store}><BrowserRouter><Header /></BrowserRouter></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('test Footer', () => {
    const tree = renderer
      .create(<Footer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
