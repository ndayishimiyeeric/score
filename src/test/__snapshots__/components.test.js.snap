// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Compontests test Footer 1`] = `
<footer>
  <p
    className="footerP"
  >
    © decathlon.com
  </p>
</footer>
`;

exports[`Test Compontests test Header 1`] = `
<header
  className="header"
>
  <a
    className="link"
    href="/"
    onClick={[Function]}
  />
  <p
    className="headerHeading"
  >
    header Heading
  </p>
  <div
    className="headerSide"
  >
    <svg
      className="sideIcon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
      />
    </svg>
    <svg
      className="sideIcon"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <desc />
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <path
        d="M10.325 4.317c.426 -1.756 2.924 -1.756 3.35 0a1.724 1.724 0 0 0 2.573 1.066c1.543 -.94 3.31 .826 2.37 2.37a1.724 1.724 0 0 0 1.065 2.572c1.756 .426 1.756 2.924 0 3.35a1.724 1.724 0 0 0 -1.066 2.573c.94 1.543 -.826 3.31 -2.37 2.37a1.724 1.724 0 0 0 -2.572 1.065c-.426 1.756 -2.924 1.756 -3.35 0a1.724 1.724 0 0 0 -2.573 -1.066c-1.543 .94 -3.31 -.826 -2.37 -2.37a1.724 1.724 0 0 0 -1.065 -2.572c-1.756 -.426 -1.756 -2.924 0 -3.35a1.724 1.724 0 0 0 1.066 -2.573c-.94 -1.543 .826 -3.31 2.37 -2.37c1 .608 2.296 .07 2.572 -1.065z"
      />
      <circle
        cx="12"
        cy="12"
        r="3"
      />
    </svg>
  </div>
</header>
`;

exports[`Test Compontests test Header 2`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    />
    <p
      className="headerHeading"
    >
      Sports
    </p>
    <div
      className="headerSide"
    >
      <svg
        className="sideIcon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
        />
      </svg>
      <svg
        className="sideIcon"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc />
        <path
          d="M0 0h24v24H0z"
          fill="none"
          stroke="none"
        />
        <path
          d="M10.325 4.317c.426 -1.756 2.924 -1.756 3.35 0a1.724 1.724 0 0 0 2.573 1.066c1.543 -.94 3.31 .826 2.37 2.37a1.724 1.724 0 0 0 1.065 2.572c1.756 .426 1.756 2.924 0 3.35a1.724 1.724 0 0 0 -1.066 2.573c.94 1.543 -.826 3.31 -2.37 2.37a1.724 1.724 0 0 0 -2.572 1.065c-.426 1.756 -2.924 1.756 -3.35 0a1.724 1.724 0 0 0 -2.573 -1.066c-1.543 .94 -3.31 -.826 -2.37 -2.37a1.724 1.724 0 0 0 -1.065 -2.572c-1.756 -.426 -1.756 -2.924 0 -3.35a1.724 1.724 0 0 0 1.066 -2.573c-.94 -1.543 .826 -3.31 2.37 -2.37c1 .608 2.296 .07 2.572 -1.065z"
        />
        <circle
          cx="12"
          cy="12"
          r="3"
        />
      </svg>
    </div>
  </header>
  <div
    className="mainDiv"
  >
    <div
      className="showcaseDiv"
    >
      <div
        className="showcaseImg"
      >
        <img
          alt="showcase"
          src="https://sports-api-production.s3.amazonaws.com/uploads/sport/icon/224/224.svg"
        />
      </div>
      <div
        className="showcaseDetails"
      >
        <p>
          SPORTS
        </p>
        <span>
          0
            games
        </span>
      </div>
    </div>
    <p
      className="heading"
    >
      SPORTS BY GROUP
    </p>
    <form
      onSubmit={[Function]}
    >
      <input
        className="formInput"
        name="textValue"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
    </form>
    <div
      className="missing"
    >
      <p>
        No such game available
      </p>
    </div>
  </div>
  <footer>
    <p
      className="footerP"
    >
      © decathlon.com
    </p>
  </footer>
</div>
`;

exports[`Test Compontests test Sports 1`] = `
Array [
  <header
    className="header"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    />
    <p
      className="headerHeading"
    >
      Sports
    </p>
    <div
      className="headerSide"
    >
      <svg
        className="sideIcon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
        />
      </svg>
      <svg
        className="sideIcon"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc />
        <path
          d="M0 0h24v24H0z"
          fill="none"
          stroke="none"
        />
        <path
          d="M10.325 4.317c.426 -1.756 2.924 -1.756 3.35 0a1.724 1.724 0 0 0 2.573 1.066c1.543 -.94 3.31 .826 2.37 2.37a1.724 1.724 0 0 0 1.065 2.572c1.756 .426 1.756 2.924 0 3.35a1.724 1.724 0 0 0 -1.066 2.573c.94 1.543 -.826 3.31 -2.37 2.37a1.724 1.724 0 0 0 -2.572 1.065c-.426 1.756 -2.924 1.756 -3.35 0a1.724 1.724 0 0 0 -2.573 -1.066c-1.543 .94 -3.31 -.826 -2.37 -2.37a1.724 1.724 0 0 0 -1.065 -2.572c-1.756 -.426 -1.756 -2.924 0 -3.35a1.724 1.724 0 0 0 1.066 -2.573c-.94 -1.543 .826 -3.31 2.37 -2.37c1 .608 2.296 .07 2.572 -1.065z"
        />
        <circle
          cx="12"
          cy="12"
          r="3"
        />
      </svg>
    </div>
  </header>,
  <div
    className="mainDiv"
  >
    <div
      className="showcaseDiv"
    >
      <div
        className="showcaseImg"
      >
        <img
          alt="showcase"
          src="https://sports-api-production.s3.amazonaws.com/uploads/sport/icon/224/224.svg"
        />
      </div>
      <div
        className="showcaseDetails"
      >
        <p>
          SPORTS
        </p>
        <span>
          0
            games
        </span>
      </div>
    </div>
    <p
      className="heading"
    >
      SPORTS BY GROUP
    </p>
    <form
      onSubmit={[Function]}
    >
      <input
        className="formInput"
        name="textValue"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
    </form>
    <div
      className="missing"
    >
      <p>
        No such game available
      </p>
    </div>
  </div>,
  <footer>
    <p
      className="footerP"
    >
      © decathlon.com
    </p>
  </footer>,
]
`;
